### DISCUSSION TITLE
I have a LaTeX Error: compilation failed- error... `expl3.sty'

### DISCUSSION THREAD

**Discussion by Mgonrub** [quarto-dev/quarto-cli#6293] [Q&A] [answered]
### Description

Hi all
I post this here as a log of compilation: 

`quarto render 2.ipynb --to beamer
pandoc
  to: beamer
  output-file: 2.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf

metadata
  block-headings: true
  title: Bar Charts con R
  theme: Berlin

running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

updating tlmgr

updating existing packages

compilation failed- error
LaTeX Error: Mismatched LaTeX support files detected.
(LaTeX)        Loading 'expl3.sty' aborted!
(LaTeX)
(LaTeX)        The L3 programming layer in the LaTeX format
(LaTeX)        is dated 2023-06-05, but in your TeX tree the files require
(LaTeX)        at least 2023-06-30.

For immediate help type H <return>.
 ...

l.77      \ExplLoaderFileDate{expl3.sty}}
                                         %

see 2.log for more information.`
I have until yesterday everything worked with this file or others. But now with no file.
Windows 10, jupyter notebook.
Please, if someone could help me. Thank you so much

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Please do note that Quarto highly recommends to use the TinyTex LaTeX distribution exactly for this reason.
To debug, if you want to keep your distribution, use `keep-tex: true` one your file, look at the tex file and try to compile using your distribution and look on the Internet for the solution for your distribution.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by cderv** [comment]
````
(LaTeX) The L3 programming layer in the LaTeX format
(LaTeX) is dated 2023-06-05, but in your TeX tree the files require
(LaTeX) at least 2023-06-30.
````

Usually this is a LaTeX installation / update issue. 

You can find similar issue on the web not quarto related 
* https://stackoverflow.com/questions/69076268/latex3-error-mismatched-latex-support-files-detected-latex3-loading-expl3
* https://tex.stackexchange.com/questions/586467/outdated-l3-programming-layer-for-expl3-sty
* https://community.rstudio.com/t/ctex-in-r-markdown-problem/122646/3

Try fixing the install or removing and reinstalling.

**Comment by Mgonrub** [comment, marked as answer]
Thank you all for your answers.

The solution for me was to uninstall quarto and reinstall quarto and update.

Again, thank you very much.

Greetings

**Comment by nextmarte** [comment]
I faced the same issue on Ubuntu and solved by removing and reinstalling tinytex, the problem hapened when i acidentaly interruped an "updating tlmgr" process while rendering a document.

**Reply by cderv** [reply]
Thanks for sharing this. Bad update could definitely be a cause !

**Reply by javajon** [reply]
Yes, I had the same error message when rendering Quarto (v 1.3.450) to PDF (the revealJS rendering was fine). To fix the problem I updated tinytext:

```
quarto install tinytex
 ? tinytex is already installed. Do you want to update to v2023.10? (Y/n) » Yes
 ```

After that the PDF is generated without error
